cmd/kubeadm/app/cmd/init.go

```diff
@@ -386,12 +386,33 @@ func newInitData(cmd *cobra.Command, args []string, options *initOptions, out io
 		// Validate that also the required kubeconfig files exists and are invalid, because
 		// kubeadm can't regenerate them without the CA Key
 		kubeconfigDir := options.kubeconfigDir
-		if options.dryRun {
-			kubeconfigDir = dryRunDir
-		}
 		if err := kubeconfigphase.ValidateKubeconfigsForExternalCA(kubeconfigDir, cfg); err != nil {
 			return nil, err
 		}
+
+		// If using an externalCA while dryrun, need to copy those files to dryrun dir for later use
+		if options.dryRun {
```

Comment:
given we have both https://github.com/kubernetes/kubernetes/blob/b70f397cde76af086e81f80a348607e750128869/cmd/kubeadm/app/cmd/init.go#L535 and https://github.com/kubernetes/kubernetes/blob/b70f397cde76af086e81f80a348607e750128869/cmd/kubeadm/app/cmd/init.go#L475 it seems more appropriate to do this in the "certs" phase.

Reply:
yeah, fixed!